Objects/typeobject.c

```diff
@@ -3947,8 +3951,8 @@ type_setattro(PyTypeObject *type, PyObject *name, PyObject *value)
     if (type->tp_flags & Py_TPFLAGS_IMMUTABLETYPE) {
         PyErr_Format(
             PyExc_TypeError,
-            "can't set attributes of built-in/extension type '%s'",
-            type->tp_name);
+            "cannot set '%R' attribute of immutable type '%s'",
```

Comment:
'%R' would add too many quotes, no?

```suggestion
            "cannot set %R attribute of immutable type '%s'",
```

Reply:
I was not aware of that :) Thanks!